src/Monolog/Handler/StreamHandler.php

```diff
@@ -99,6 +110,25 @@ protected function write(array $record)
 
     private function customErrorHandler($code, $msg)
     {
-        $this->errorMessage = preg_replace('{^fopen\(.*?\): }', '', $msg);
+        $this->errorMessage = preg_replace('{^(fopen|mkdir)\(.*?\): }', '', $msg);
+    }
+
+    /**
+     * @param string $stream
+     *
+     * @return null|string
+     */
+    private function getDirFromStream($stream)
+    {
+        $pos = strpos($stream, '://');
+        if ($pos === false) {
+            return dirname($stream);
+        }
+
+        if ('file:///' === substr($stream, 0, 8)) {
+            return DIRECTORY_SEPARATOR.dirname(substr($stream, 8));
```

Comment:
@Seldaek is there really a filemapper file://, cause in my tests i only get file:/// running?